game/CMakeLists.txt

```diff
@@ -60,6 +60,46 @@ endif()
 
 set(SOURCES ${SOURCE_FILES} ${HEADER_FILES} ${RESOURCE_FILES})
 
+# Define main executable
+add_executable(performous ${SOURCES})
+
+  string(ASCII 27 Esc)
+# Enable C++14 support and other compiler flags...
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fcx-limited-range ")
+	execute_process(COMMAND ${CMAKE_CXX_COMPILER} -dumpversion OUTPUT_VARIABLE GCC_VERSION)
+	message("Detected ${Esc}[36mGCC ${Esc}[mversion ${Esc}[36m${GCC_VERSION}${Esc}[m")
+	if (GCC_VERSION VERSION_GREATER 8 OR GCC_VERSION VERSION_EQUAL 8)
+		message(STATUS "Setting C++ standard to: ${Esc}[36mC++17${Esc}[m")
+		set_property(TARGET performous PROPERTY CXX_STANDARD 17)
+	elseif (GCC_VERSION VERSION_GREATER 7 OR GCC_VERSION VERSION_EQUAL 7)
+		message(STATUS "Setting C++ standard to: ${Esc}[36mC++14${Esc}[m")
+		set_property(TARGET performous PROPERTY CXX_STANDARD 14)
+	elseif (GCC_VERSION VERSION_GREATER 4.8 OR GCC_VERSION VERSION_EQUAL 4.8)
+		set_property(TARGET performous PROPERTY CXX_STANDARD 11)
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-attributes -Wno-deprecated-declarations")
+		message(WARNING "Your GCC is rather old, disabling some useful compiler warnings (GCC 5+ or Clang 3.4+ recommended for proper C++14 support).")
+	else()
+		message(FATAL_ERROR "Your C++ compiler is too old, please upgrade.")
+	endif()
+endif()
+
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+	message("Detected ${Esc}[36mClang/AppleClang ${Esc}[mversion ${Esc}[36m${CMAKE_CXX_COMPILER_VERSION}${Esc}[m")
+	if(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" OR CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 5.0)
+		message(STATUS "Setting C++ standard to: ${Esc}[36mC++17${Esc}[m")
+		set_property(TARGET performous PROPERTY CXX_STANDARD 17)
+	elseif (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 3.4 )
+		message(STATUS "Setting C++ standard to: C${Esc}[36m++14${Esc}[m")
+		set_property(TARGET performous PROPERTY CXX_STANDARD 14)
+	else ()
+	# FIXME: This case is probably not correct but seriously, I don't think it will come up very frequently anyway.
+		set_property(TARGET performous PROPERTY CXX_STANDARD 11)
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-attributes -Wno-deprecated-declarations")
+		message(WARNING "Your GCC is rather old, disabling some useful compiler warnings (GCC 5+ or Clang 3.4+ recommended for proper C++14 support).")
+	endif()
+endif()
+
```

Comment:
I wouldn't do this within this pr.
Always try to create as small as possible pr's

Regarding the output of this pr it should be buildable, fast as the cmakelists file is righ tnow

Reply:
I can split it if you want but I jumped onto this particular bandwagon because of the initial failure to build on mac, which was a direct consequence of my cleaning and updating the mac building script (which did make sense in this PR)

Reply:
Hmm well actually the build script itself didn't need an update, since it used to work with my pr aswell? 🤔 
Granted it has some tweaks which probably would help but i think it's better to do it within another pr